### DISCUSSION TITLE
How to use form object to submit relationship table values?

### DISCUSSION THREAD

**Discussion by lianmaymesi** [livewire/livewire#8207] [1. Help] [answered]
In some scenarios, we must submit the form and the relationship table values. I understood how to use the form object, and I used to follow the below pattern. But, I don't know how possible this pattern is livewire standard or not. Because the properties we use in `ProductForm.php` are only the `Product.php` model attributes and cannot use the `ProductVariation.php` model attributes.

```php
public Product $product;

public $price;
public $title;

public ProductVariation $variation;

public function create()
{
     $this->validate([
          'price' => 'required',
          'title' => 'required',
          'variation' => 'required'
     ]);

     $product = Product::create([...];
     $product->variation()->create([...]);
}

public function setProduct(Product $product = null)
{
     $this->price = $product->price;
     $this->title = $product->title;
     $this->variation = $product->variation;
}
```

**Comment by joshhanley** [comment, marked as answer]
@lianmaymesi your form object should be a representation of the form that is being displayed to the user. So while it might be called ProductForm, if the browser is displaying inputs for editing both Product and ProductVariation fields, then your form object should also do the same.

So I wouldn't assign the variation as an eloquent model, instead I would split it out into it's attributes into properties too. Then in create, you can pass product properties into `Product::create()` and do the same for the variation.

Hope this helps!
